src/Server/Sanitizer.fs

```diff
@@ -13,59 +13,186 @@ let unsafe = ["System"; "NativePtr"]
 let isLongIdentIncludeUnsafe (ident: LongIdent) =
   ident |> Seq.exists (fun i -> List.contains i.idText unsafe)
 
+let isTypeInheritUnsafe = function
+  | SynType.LongIdent (LongIdentWithDots(ident, _)) ->
+    ident |> isLongIdentIncludeUnsafe
+  | SynType.LongIdentApp (_, LongIdentWithDots(ident, _), _, _, _, _, _) ->
+    ident |> isLongIdentIncludeUnsafe
+  | _ -> false
+
 let rec visitExpr = function
+  // Application
   | SynExpr.App (_, _, f, arg, _) ->
     visitExpr f
     || visitExpr arg
+
+  // Long Identifier
   | SynExpr.LongIdent (_, LongIdentWithDots (ident, _), _, _) ->
     ident |> isLongIdentIncludeUnsafe
-  | SynExpr.Match (_, e, clauses, _) ->
+
+  // Match expressions
+  | SynExpr.Match (_, e, clauses, _)
+  | SynExpr.MatchBang (_, e, clauses, _) ->
     visitExpr e
-    || clauses |> List.fold (fun acc (Clause (_, wh, e, _, _)) ->
-      acc
-      || visitExpr e
-      || (match wh with | Some e -> visitExpr e | None -> false)) false
+    || clauses |> List.fold visitClause false
+  | SynExpr.MatchLambda (_, _, clauses, _, _) ->
+    clauses |> List.fold visitClause false // Maybe List.exists?
```

Comment:
forall

Reply:
We would need List.exists here, since `visitExpr` returns true on an unsafe match and false on no match. We should return true if there is at least one true.

Reply:
Very good. I thought the opposite way. List.exists looks better to me.